homeassistant/components/lgthinq/entity.py

```diff
@@ -0,0 +1,119 @@
+"""Base class for ThinQ entities."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from thinqconnect import ThinQAPIErrorCodes, ThinQAPIException
+from thinqconnect.integration.homeassistant.property import Property as ThinQProperty
+
+from homeassistant.core import callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import DeviceDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(kw_only=True)
+class PropertyInfo:
+    """A data class contains an information for creating property."""
+
+    # The property key for use in SDK must be snake_case string.
+    key: str
+
+    # Optional, if the value should be converted before calling api.
+    value_converter: Callable[[Any], Any] | None = None
+
+
+@dataclass(kw_only=True, frozen=True)
+class ThinQEntityDescription(EntityDescription):
+    """The base thinq entity description."""
+
+    property_info: PropertyInfo
+
+
+class ThinQEntity(CoordinatorEntity[DeviceDataUpdateCoordinator]):
+    """The base implementation of all lg thinq entities."""
+
+    _attr_has_entity_name = True
+    entity_description: ThinQEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: ThinQEntityDescription,
+        property: ThinQProperty,
+    ) -> None:
+        """Initialize an entity."""
+        super().__init__(coordinator)
+
+        self.entity_description = entity_description
+        self._property = property
+        self._attr_device_info = coordinator.device_info
+
+        # Set the unique key. If there exist a location, add the prefix location name.
+        unique_key = (
+            f"{entity_description.key}"
+            if property.location is None
+            else f"{property.location}_{entity_description.key}"
+        )
```

Comment:
What's a location?

Reply:
In case of a refrigerator, 'location' is one of 'freezer room' and 'refrigerator room'
'Target temperature' for those two rooms has same key.
That's why we create unique id with 'location'.

Reply:
check